### PR TITLE
Fix cache key collision when 2 tables have same list of entity

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:

Currently cache key in standard transformer is constructed by entity values and project name. However two or more feature retrieval table can have same for both values which cause cache key collision.
This PR also do some refactoring in feature_retriever.go

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally

### CODE REVIEW

File: api/pkg/transformer/feast/feature_retriever.go [hunk 1]
```diff
@@ -95,6 +92,7 @@ type Options struct {
 	BatchSize               int           `envconfig:"FEAST_BATCH_SIZE" default:"50"`
 	CacheEnabled            bool          `envconfig:"FEAST_CACHE_ENABLED" default:"true"`
 	CacheTTL                time.Duration `envconfig:"FEAST_CACHE_TTL" default:"60s"`
+	CacheSizeInMB           int           `envconfig:"CACHE_SIZE_IN_MB" default:"100"`
```
Comment (ariefrahmansyah) [inline api/pkg/transformer/feast/feature_retriever.go:95]: I prefer `FEAST_CACHE_SIZE_IN_MB` as env name

File: api/pkg/transformer/feast/feature_retriever.go [hunk 2]
```diff
@@ -529,90 +340,18 @@ func getColumnNames(config *spec.FeatureTable) []string {
...
-			results = append(results, base64.StdEncoding.EncodeToString(bytes))
-		}
-		return results, types.ValueType_STRING_LIST, nil
-	default:
-		return nil, types.ValueType_INVALID, fmt.Errorf("unknown feature cacheValue type: %T", val.Val)
-	}
+	return entitySet
 }
 
 func mergeColumnTypes(dst []types.ValueType_Enum, src []types.ValueType_Enum) []types.ValueType_Enum {
```
Comment (ariefrahmansyah) [inline api/pkg/transformer/feast/feature_retriever.go:357]: nit: move this function to `types.go`

**Review by ariefrahmansyah — APPROVED** [review]
LGTM with nit comments.

File: api/pkg/transformer/feast/feature_retriever.go (hunk 1 above)
Comment (pradithya) [reply api/pkg/transformer/feast/feature_retriever.go:95]: It will be a breaking change if we rename the env var.

**Pull request merged**